programs/odbc-bridge/ColumnInfoHandler.cpp

```diff
@@ -160,7 +160,15 @@ void ODBCColumnsInfoHandler::handleRequest(HTTPServerRequest & request, HTTPServ
         }
 
         WriteBufferFromHTTPServerResponse out(response, request.getMethod() == Poco::Net::HTTPRequest::HTTP_HEAD, keep_alive_timeout);
-        writeStringBinary(columns.toString(), out);
+        try
+        {
+            writeStringBinary(columns.toString(), out);
+            out.finalize();
+        }
+        catch (...)
+        {
+            out.finalize();
```

Comment:
What did you try to achieve by it?

Reply:
It is hard to remember, but likely if the `writeStringBinary` was triggered flush, then it may throw, so we need to call `finalize` reset the `pos` so that `next` will be no-op (note, that `cancel` was not there in 2021).